static/js/unread_ops.js

```diff
@@ -22,6 +22,9 @@ export function mark_all_as_read() {
         success: () => {
             // After marking all messages as read, we reload the browser.
             // This is useful to avoid leaving ourselves deep in the past.
+            // This is also the desired behavior in case of partial success,
+            // where the request times out after marking some messages as read,
+            // so we don't need to distinguish that scenario here.
```

Comment:
Is it?  Take the extreme case where for some reason only one message is marked as read before the timeout.  If I, as a user, clicked a button that said "mark all messages as read," and waited the 50s, and the app reloaded with one fewer unread message, would that be satisfactory?

I think we need a retry loop on 206 -- ideally along with some user-visible "please wait, flipping bits on your messages," though that is probably out of scope for this PR.

Reply:
Yeah, though I'm not sure how well the frontend handles these mark-all-as-read situation in terms of refreshing properly in response to events received - given that in the case of full success, we do need to reload, as mentioned in the comment here:
```
            // After marking all messages as read, we reload the browser.
            // This is useful to avoid leaving ourselves deep in the past.
```
I'm not sure how that should be different in case of these partial successes - though I admit I don't quite fully understand the frontend issues here that are making us have to reload to begin with

I tweaked the commit at least though, to call this a `currently intended behavior` rather than `desired` and added a `TODO` note on how this ideally would be improved.